Source/Core/DolphinQt2/GameList/GameList.cpp

```diff
@@ -105,18 +115,47 @@ void GameList::MakeListView()
 
 void GameList::ShowContextMenu(const QPoint&)
 {
+  const auto game = GetSelectedGame();
+  if (game.isEmpty())
+    return;
+
   QMenu* menu = new QMenu(this);
-  DiscIO::Platform platform = GameFile(GetSelectedGame()).GetPlatformID();
+  DiscIO::Platform platform = GameFile(game).GetPlatformID();
+  menu->addAction(tr("Properties"), this, SLOT(OpenProperties()));
+  menu->addAction(tr("Wiki"), this, SLOT(OpenWiki()));
+  menu->addSeparator();
+
   if (platform == DiscIO::Platform::GAMECUBE_DISC || platform == DiscIO::Platform::WII_DISC)
   {
-    menu->addAction(tr("Properties"), this, SLOT(OpenProperties()));
-    menu->addAction(tr("Open Wiki Page"), this, SLOT(OpenWiki()));
-    menu->addAction(tr("Set as Default ISO"), this, SLOT(SetDefaultISO()));
+    menu->addAction(tr("Default ISO"), this, SLOT(SetDefaultISO()));
+    const auto blob_type = GameFile(game).GetBlobType();
+
+    if (blob_type == DiscIO::BlobType::GCZ)
+      menu->addAction(tr("Decompress ISO"), this, SLOT(DecompressISO()));
+    else if (blob_type == DiscIO::BlobType::PLAIN)
+      menu->addAction(tr("Compress ISO"), this, SLOT(CompressISO()));
+
+    menu->addSeparator();
   }
-  else
+  if (platform == DiscIO::Platform::WII_WAD)
   {
-    return;
+    if (GameFile(game).IsInstalled())
+      menu->addAction(tr("Uninstall from NAND"), this, SLOT(UninstallWAD()));
+    else
+      menu->addAction(tr("Install to NAND"), this, SLOT(InstallWAD()));
```

Comment:
This should be enabled *even when the title is already installed*. This is because we only check whether the title is installed, not if the exact same version is installed.

The user should also be allowed to easily reinstall a title from a WAD in case they have corrupted their title content files, which can happen for developers, people trying stuff on their NAND, or just because Dolphin sucks at importing titles properly and [still does so incorrectly](https://bugs.dolphin-emu.org/issues/10170).

Please also put "Install" above Uninstall in any case.